orientdb/src/test/java/org/hibernate/datastore/ogm/orientdb/utils/OrientDBTestHelper.java

```diff
@@ -0,0 +1,119 @@
+
+	private static final String JDBC_URL = "jdbc:orient:memory:test";
+	private static Map<String, List<ORecordId>> classIdMap;
+
+	public OrientDBTestHelper() {
+		BasicConfigurator.configure();
+		// create OrientDB in memory
+		org.hibernate.datastore.ogm.orientdb.util.MemoryDBUtil.createDbFactory( "jdbc:orient:memory:test" );
+	}
+
+	@Override
+	public long getNumberOfEntities(SessionFactory sessionFactory) {
+		System.out.println( "org.hibernate.datastore.ogm.orientdb.util.OrientDBTestHelper.getNumberOfEntities()" );
+		long summ = 0;
+		for ( Map.Entry<String, List<ORecordId>> entry : classIdMap.entrySet() ) {
+			String key = entry.getKey();
+			List<ORecordId> value = entry.getValue();
+			for ( int i = 0; i < value.size(); i++ ) {
+				ORecordId get = value.get( i );
+				summ++;
+			}
+
+		}
+		return classIdMap.size();
+	}
+
+	@Override
+	public long getNumberOfAssociations(SessionFactory sessionFactory) {
+		System.out.println( "org.hibernate.datastore.ogm.orientdb.util.OrientDBTestHelper.getNumberOfAssociations()" );
+		throw new UnsupportedOperationException( "Not supported yet." );
+	}
+
+	@Override
+	public long getNumberOfAssociations(SessionFactory sessionFactory, AssociationStorageType type) {
+		System.out.println( "org.hibernate.datastore.ogm.orientdb.util.OrientDBTestHelper.getNumberOfAssociations()" );
+		throw new UnsupportedOperationException( "Not supported yet." );
+	}
+
+	@Override
+	public Map<String, Object> extractEntityTuple(SessionFactory sessionFactory, EntityKey key) {
+		System.out.println( "org.hibernate.datastore.ogm.orientdb.util.OrientDBTestHelper.extractEntityTuple()" );
+		throw new UnsupportedOperationException( "Not supported yet." );
+	}
+
+	@Override
+	public boolean backendSupportsTransactions() {
+		System.out.println( "org.hibernate.datastore.ogm.orientdb.util.OrientDBTestHelper.backendSupportsTransactions()" );
+		return true;
+	}
+
+	@Override
+	public void dropSchemaAndDatabase(SessionFactory sessionFactory) {
+		System.out.println( "org.hibernate.datastore.ogm.orientdb.util.OrientDBTestHelper.dropSchemaAndDatabase()" );
+
+		Map<String, ClassMetadata> metadata = sessionFactory.getAllClassMetadata();
+		for ( Map.Entry<String, ClassMetadata> entry : metadata.entrySet() ) {
+			String key = entry.getKey();
+			ClassMetadata value = entry.getValue();
+			System.err.println( "key: " + key );
+			System.err.println( "value: " + value );
+		}
+
+	}
+
+	@Override
+	public Map<String, String> getEnvironmentProperties() {
+		System.out.println( "org.hibernate.datastore.ogm.orientdb.util.OrientDBTestHelper.getEnvironmentProperties()" );
+		Map<String, String> properties = new HashMap<String, String>();
+		properties.put( "javax.persistence.jdbc.url", JDBC_URL );
+		properties.put( "javax.persistence.jdbc.driver", "com.orientechnologies.orient.jdbc.OrientJdbcDriver" );
+		properties.put( "javax.persistence.jdbc.user", "admin" );
+		properties.put( "javax.persistence.jdbc.password", "admin" );
```

Comment:
In the end Hibernate OGM does not use the JDBC driver like Hibernate (in general).
OgmProperties should contains the one to set for OGM.
For this case in particolar, being embedded, the only properties that the user should be force to set is the dialect.

Check the other dialect examples.